src/main/java/randoop/reflection/SignatureParser.java

```diff
@@ -0,0 +1,154 @@
+package randoop.reflection;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** Parses type signature strings used to identify methods and constructors in input. */
+// TODO: This duplicates should be factored into a separate source set (aka, module) so that it
+// can also be used in javagents. The patterns are duplicated from {@code ReplacementFileReader}
+// from the mapcall agent.
+public class SignatureParser {
+  /** Regex for Java identifiers */
+  private static final String ID_STRING =
+      "\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*";
+
+  /**
+   * Regex to match a sequence of identifiers (or {@code <init>}) separated by periods. Corresponds
+   * to package names, fully-qualified classnames, or method names with fully-qualified classname.
+   */
+  private static final String DOT_DELIMITED_IDS =
+      ID_STRING + "(?:\\." + ID_STRING + ")*" + "(?:\\.<init>)?";
+
+  /**
+   * Naive regex to match a method signature consisting of a fully-qualified method name followed by
+   * anything in parentheses. The parentheses are expected to contain argument types, but the
+   * pattern permits anything.
+   *
+   * <p>Capturing group 1 matches the fully-qualified method name, and capturing group 2 matches the
+   * contents of the parentheses.
+   */
+  private static final Pattern SIGNATURE_PATTERN =
+      Pattern.compile("(" + DOT_DELIMITED_IDS + ")\\(([^)]*)\\)");
+
+  /**
+   * Parses a fully-qualified signature and returns the corresponding {@code
+   * java.lang.reflect.AccessibleObject}.
+   *
+   * <p>A signature is expected to have the form
+   *
+   * <ul>
+   *   <li>{@code package-name.classname.method-name(argument-list)} for a method,
+   *   <li>{@code package-name.classname.&lt;init&gt;(argument-list)} or {@code
+   *       package-name.classname(argument-list)} for a constructor.
+   * </ul>
+   *
+   * <p>where {@code package-name} is a period-separated list of identifiers, and <code>
+   * argument-list</code> is a comma-separated (spaces-allowed) list of fully-qualified Java raw
+   * types. Array types have the format <code>element-type[]</code>.
+   *
+   * @param signature the string to parse: a signature string for a method or constructor, in the
+   *     above format
+   * @param visibility the predicate for determining whether the method or constructor is visible
+   * @param reflectionPredicate the predicate for checking reflection policy
+   * @return the {@code AccessibleObject} for the method or constructor represented by the string
```

Comment:
It can also return null.  Document that fact, and under what circumstances null is returned.